Source/Core/DolphinQt2/Main.cpp

```diff
@@ -8,17 +8,49 @@
 #include <QMessageBox>
 #include <QObject>
 
+#include "Common/MsgHandler.h"
 #include "Core/Analytics.h"
 #include "Core/BootManager.h"
 #include "Core/Core.h"
 #include "DolphinQt2/Host.h"
 #include "DolphinQt2/InDevelopmentWarning.h"
 #include "DolphinQt2/MainWindow.h"
+#include "DolphinQt2/QtUtils/RunOnObject.h"
 #include "DolphinQt2/Resources.h"
 #include "DolphinQt2/Settings.h"
 #include "UICommon/CommandLineParse.h"
 #include "UICommon/UICommon.h"
 
+bool QtMsgAlertHandler(const char* caption, const char* text, bool yes_no, MsgType style)
+{
+  return RunOnObject(QApplication::instance(), [&] {
+    QMessageBox message_box(QApplication::activeWindow());
+    message_box.setWindowTitle(QString::fromUtf8(caption));
+    message_box.setText(QString::fromUtf8(text));
+    message_box.setStandardButtons(yes_no ? QMessageBox::Yes | QMessageBox::No : QMessageBox::Ok);
+
+    QMessageBox::Icon icon = QMessageBox::NoIcon;
+    switch (style)
```

Comment:
To avoid an reassignment, you can do: 

```
const QMessage::Icon icon = [&style] {
switch(style) {
  case INFORMATION: 
    return QMessageBox::Information;
  case QUESTION:
    return QMessageBox::Question;
  case WARNING:
    return QMessageBox::Warning;
   case CRITICAL:
    return QMessageBox::Critical;
   default:
    return QMessageBox::NoIcon;
}
}();
```

you can even use it in the `setIcon` function but I'm sure the compiler will handle it for you